### PR TITLE
Avoid downloading redundant email and pass data as d-bus signal parameter

### PR DESCRIPTION
**Pull request by g33kyaditya**
The last two different commits are of interest.

One in which we avoid downloading new email if it's not a new one.

In the other, we pass the restaurant plugin information in QStringLists over D-Bus signal

### CODE REVIEW

File: plugins/restaurantreservation/restaurantreservation.cpp [hunk 1]
```diff
@@ -143,17 +143,30 @@ void RestaurantReservation::initDatabase()
 
 void RestaurantReservation::setDBusData()
```
Comment (Ashish-Bansal) [inline plugins/restaurantreservation/restaurantreservation.cpp]: Change function name to sendDataOverDbus.

File: plugins/restaurantreservation/restaurantreservation.cpp [hunk 2]
```diff
@@ -143,17 +143,30 @@ void RestaurantReservation::initDatabase()
...
+    values.append(QString::number(m_partySize));
+    values.append(m_restaurantName);
+    values.append(m_streetAddress);
+    values.append(m_addressLocality);
+    values.append(m_addressRegion);
+
+    emit update(keys, values);
 }
 
 void RestaurantReservation::getDataFromDatabase()
```
Comment (Ashish-Bansal) [inline plugins/restaurantreservation/restaurantreservation.cpp]: getDataFromDatabase suggests that it's a getter but it isn't really. It should return listOfMapsInDatabase.

**Comment by g33kyaditya** [comment]
Built and Tested. Works as expected. The changes for review are in restaurant plugin

The way sync process works is that as soon as daemon starts it will load all plugins. Once the plugin is loaded and records from database are fetched and stored in a m_listOfMapsInDatabase , a signal is emitted over d-bus. On receiving this signal, the plasmoid will call a d-bus method on plugin side. This method, then sends all data one by one by passing the keys,values as signal parameters of d-bus signal.

Changes needed to be done: On plasmoid side, check for duplication of cards.

File: src/daemon.cpp [hunk 3]
```diff
@@ -31,6 +31,9 @@ Daemon::Daemon(QObject* parent): QObject(parent)
     QDBusConnection dbus = QDBusConnection::sessionBus();
     dbus.registerObject("/KDENow", this);
     dbus.registerService("org.kde.kdenow");
+    Processor * processor = SingletonFactory::instanceFor< Processor >();
+    connect(this, &Daemon::fetchRecordsFromDatabase, processor, &Processor::loadPlugins);
```
Comment (Ashish-Bansal) [inline src/daemon.cpp]: No need to connect it via signal and emit the signal in next line. 

Just do, processor->loadPlugins();

File: src/processor.h [hunk 4]
```diff
@@ -33,10 +33,10 @@ class KDENOWCORE_EXPORT Processor : public QObject
 
     public:
         explicit Processor(QObject* parent = 0);
-        void loadPlugins();
 
     public Q_SLOTS:
         void process(KIMAP::MessagePtr messagePtr);
+        void loadPlugins();
```
Comment (Ashish-Bansal) [inline src/processor.h]: Move it back as public member function.

File: src/daemon.h [hunk 5]
```diff
@@ -36,8 +36,11 @@ class KDENOWCORE_EXPORT Daemon : public QObject
         Daemon(QObject* parent = 0);
         void startEmailManagers();
 
+    Q_SIGNALS:
```
Comment (Ashish-Bansal) [inline src/daemon.h]: Remove this signal.

**Pull request merged**